tree.go

```diff
@@ -247,6 +248,9 @@ func (n *node) addRoute(path string, handlers HandlersChain) {
 func (n *node) insertChild(numParams uint8, path string, fullPath string, handlers HandlersChain) {
 	var offset int // already handled bytes of the path
 
+	// save the node's full path
+	var nodeFullPath string
```

Comment:
Thank you for taking in consideration my tip. But do you need this `nodeFullPath`?
`fullPath` parameter is ok i think. You add it at the end.

Reply:
i want to give every node a fullpath value,not just the end node

Reply:
Using the `fullPath` you are still giving a **fullPath** to all the nodes that have handlers. A request will never hit an intermediary node, so those nodes are used only internally form the router.